detox/src/devices/drivers/android/genycloud/GenyCloudDeviceAllocator.js

```diff
@@ -0,0 +1,72 @@
+const AndroidDeviceAllocator = require('../AndroidDeviceAllocator');
+const retry = require('../../../../utils/retry');
+const logger = require('../../../../utils/logger').child({ __filename });
+
+class GenyCloudDeviceAllocator extends AndroidDeviceAllocator {
+  constructor(deviceRegistry, instanceLookupService, instanceLifecycleService) {
+    super(deviceRegistry, logger);
+
+    this.instanceLookupService = instanceLookupService;
+    this.instanceLifecycleService = instanceLifecycleService;
+  }
+
+  async _doAllocateDevice(recipe) {
+    let { instance, isNew } = await this._doSynchronizedAllocation(recipe);
+
+    instance = await this._waitForInstanceBoot(instance);
+    instance = await this._adbConnectIfNeeded(instance);
+    return {
+      instance,
+      isNew,
+      toString: () => `GenyCloud:${instance.name} (${instance.uuid})`,
```

Comment:
See my comment regarding `EmulatorHandle`, maybe we could use a sort of `GenycloudInstanceHandle`?